gherkin-utils/java/src/test/java/io/cucumber/gherkin/util/PrettyPrintGherkinDocumentTest.java

```diff
@@ -0,0 +1,283 @@
+package io.cucumber.gherkin.util;
+
+import static org.junit.Assert.assertEquals;
+
+import io.cucumber.gherkin.GherkinDocumentBuilder;
+import io.cucumber.gherkin.Parser;
+import io.cucumber.messages.IdGenerator;
+import io.cucumber.messages.types.GherkinDocument;
+import org.junit.Before;
+import org.junit.Test;
+
+public class PrettyPrintGherkinDocumentTest {
+
+    Parser<GherkinDocument> parser;
+
+    @Before
+    public void beforeEach() {
+        IdGenerator idGen = new IdGenerator.Incrementing();
+        parser = new Parser<>(new GherkinDocumentBuilder(idGen));
+    }
+
+    @Test
+    public void emptyFile() {
+        GherkinDocument gherkinDocument = parser.parse("");
+        assertEquals("", PrettyPrintGherkinDocument.prettyPrint(gherkinDocument, Syntax.gherkin));
+    }
+
+    @Test
+    public void scenarioOutlineWithExamples() {
```

Comment:
To make these test classes more readable, we could split the cases into files before/after, then assert on that, which means we don't have to escape / deal with long strings in this test class